src/pages/Governance/ProposalList/CreateProposalModal/index.tsx

```diff
@@ -200,21 +280,30 @@ export const CreateProposal: React.FC<CreateProposalProps> = ({
             <Form>
               <CurrentFields />
 
-              {currentStep === steps.length - 1 ? (
+              {proposalMode === 'file' && currentStep === 1 && (
                 <FormikSubmitButton
                   enabledLabel={t('vote.createProposalForm.create')}
                   fullWidth
                   loading={isCreateProposalLoading}
                 />
-              ) : (
-                <PrimaryButton
-                  fullWidth
-                  disabled={getErrorsByStep(currentStep)}
-                  onClick={() => setCurrentStep(currentStep + 1)}
-                >
-                  {buttonText}
-                </PrimaryButton>
               )}
+
+              {proposalMode === 'manual' &&
+                (currentStep === steps.length - 1 ? (
+                  <FormikSubmitButton
+                    enabledLabel={t('vote.createProposalForm.create')}
+                    fullWidth
+                    loading={isCreateProposalLoading}
+                  />
+                ) : (
+                  <PrimaryButton
+                    fullWidth
+                    disabled={getErrorsByStep(currentStep)}
+                    onClick={() => setCurrentStep(currentStep + 1)}
+                  >
+                    {buttonText}
+                  </PrimaryButton>
+                ))}
             </Form>
```

Comment:
I think this logic is becoming misleading: `CurrentFields` would imply to me that it only contains the form fields, but it could actually contain the initial UI which contains the two buttons to choose what flow to follow. The logic for displaying the submit button is also getting quite complicate in my opinion.

Thinking about it, I'd follow a different approach and instead make every step into a component. Coupled with nested routing, we could get rid of a lot of extra logic while being able to share common steps between both flows (the initial UI and the final step showing a preview of the proposal).